### PR TITLE
Replace JSHint with ESLint

### PR DESCRIPTION
**Pull request by pdehaan**
```sh
$ npm run lint

> vinyl@1.0.0 lint /Users/pdehaan/dev/github/vinyl
> eslint . && jscs *.js lib/ test/


/Users/pdehaan/dev/github/vinyl/index.js
  174:1  warning  Line 174 exceeds the maximum line length of 80  max-len

/Users/pdehaan/dev/github/vinyl/test/File.js
  390:45  warning  This function has too many statements (20). Maximum allowed is 15  max-statements
  668:11  warning  "a" is defined but never used                                      no-unused-vars
  680:11  warning  "a" is defined but never used                                      no-unused-vars
  712:11  warning  "a" is defined but never used                                      no-unused-vars
  756:11  warning  "a" is defined but never used                                      no-unused-vars
  800:11  warning  "a" is defined but never used                                      no-unused-vars

✖ 7 problems (0 errors, 7 warnings)


$ echo $?
0
```

### CODE REVIEW

**Comment by pdehaan** [comment]
Whitespace agnostic diff at https://github.com/gulpjs/vinyl/pull/66/files?w=0

File: test/File.js [hunk 1]
```diff
@@ -236,13 +236,13 @@ describe('File', function() {
 
       var file = new File(options);
 
-      var copy1 = file.clone({ contents: false });
+      var copy1 = file.clone({contents: false});
```
Comment (phated) [inline test/File.js]: can we adjust this rule?  I think this pattern looks terrible compared to the one we currently use.

Comment (pdehaan) [inline test/File.js]: The rule in question is [`object-curly-spacing`](http://eslint.org/docs/rules/object-curly-spacing), found in [**gulpjs/eslint-config-gulp** /eslint.json:23](https://github.com/gulpjs/eslint-config-gulp/blob/38080cc98d9d9129f122525872774043ec688444/eslint.json#L23)

Comment (pdehaan) [inline test/File.js]: @phated https://github.com/gulpjs/eslint-config-gulp/pull/4

**Comment by pdehaan** [comment]
Done. Bumped to **eslint-config-gulp@2** and adjusted all the `{` and `}` inconsistencies.

**Pull request merged**